tests/text/test_ter.py

```diff
@@ -0,0 +1,187 @@
+from functools import partial
+from typing import Sequence
+
+import pytest
+from torch import Tensor, tensor
+
+from tests.text.helpers import INPUT_ORDER, TextTester
+from torchmetrics.functional.text.ter import ter
+from torchmetrics.text.ter import TER
+from torchmetrics.utilities.imports import _SACREBLEU_AVAILABLE
+
+if _SACREBLEU_AVAILABLE:
+    from sacrebleu.metrics import TER as SacreTER
+
+# example taken from
+# https://www.nltk.org/api/nltk.translate.html?highlight=bleu%20score#nltk.translate.bleu_score.corpus_bleu and adjusted
+# EXAMPLE 1
+HYPOTHESIS_A = "It is a guide to action which ensures that the military always obeys the commands of the party"
+REFERENCE_1A = "It is a guide to action that ensures that the military will forever heed Party commands"
+REFERENCE_2A = "It is a guiding principle which makes the military forces always being under the command of the Party"
+
+# EXAMPLE 2
+HYPOTHESIS_B = "he read The Book because he was interested in World history"
+REFERENCE_1B = "he was interested in world history because he read the book"
+REFERENCE_2B = "It is the practical guide for the army always to heed the directions of the party"
+
+# EXAMPLE 3 (add intentionally whitespaces)
+HYPOTHESIS_C = "the cat the   cat on the mat "
+REFERENCE_1C = "the  cat is     on the mat "
+REFERENCE_2C = "there is a   cat on the mat"
+
+TUPLE_OF_REFERENCES = (
+    ((REFERENCE_1A, REFERENCE_2A), (REFERENCE_1B, REFERENCE_2B)),
+    ((REFERENCE_1B, REFERENCE_2B), (REFERENCE_1C, REFERENCE_2C)),
+)
+TUPLE_OF_HYPOTHESES = ((HYPOTHESIS_A, HYPOTHESIS_B), (HYPOTHESIS_B, HYPOTHESIS_C))
+
+BATCHES = {"preds": TUPLE_OF_HYPOTHESES, "targets": TUPLE_OF_REFERENCES}
+
+
+def sacrebleu_ter_fn(
+    targets: Sequence[Sequence[str]],
+    preds: Sequence[str],
+    normalized: bool,
+    no_punct: bool,
+    asian_support: bool,
+    case_sensitive: bool,
+) -> Tensor:
+    sacrebleu_ter = SacreTER(
+        normalized=normalized, no_punct=no_punct, asian_support=asian_support, case_sensitive=case_sensitive
+    )
+    # Sacrebleu CHRF expects different format of input
+    targets = [[target[i] for target in targets] for i in range(len(targets[0]))]
+    sacrebleu_ter = sacrebleu_ter.corpus_score(preds, targets).score / 100
+    return tensor(sacrebleu_ter)
+
+
+@pytest.mark.parametrize(
+    ["normalize", "no_punctuation", "asian_support", "lowercase"],
+    [
+        pytest.param(False, False, False, False),
+        pytest.param(True, False, False, False),
+        pytest.param(False, True, False, False),
+        pytest.param(False, False, True, False),
+        pytest.param(False, False, False, True),
+        pytest.param(True, True, True, True),
```

Comment:
```suggestion
        (False, False, False, False),
        (True, False, False, False),
        (False, True, False, False),
        (False, False, True, False),
        (False, False, False, True),
        (True, True, True, True),
```
unless you need some pytest kwargs like skipif or reason, tuple is fine and shorter